package.py

```diff
@@ -730,9 +730,19 @@ def commands_step(path, commands):
                         else:
                             pip_requirements_step(pip_requirements, prefix,
                                                   required=True)
+
                     if path:
                         step('zip', path, prefix)
-                        hash(path)
+                        if patterns:
+                            # Take patterns into account when computing hash
+                            pf = ZipContentFilter(args=self._args)
+                            pf.compile(patterns)
+
+                            for path in pf.filter(path, prefix):
```

Comment:
We may want to use another name than `path` to avoid confusion.

Reply:
How about `path_from_pattern`?

Reply:
Done.